controllers/virtinkmachine_controller.go

```diff
@@ -120,13 +174,11 @@ func (r *VirtinkMachineReconciler) reconcile(ctx context.Context, machine *infra
 
 		vm.Name = vmKey.Name
 		vm.Namespace = vmKey.Namespace
-		if err := controllerutil.SetControllerReference(machine, vm, r.Scheme); err != nil {
-			return fmt.Errorf("set VM controller reference: %s", err)
-		}
-		if err := r.Create(ctx, vm); err != nil {
-			return fmt.Errorf("create VM: %s", err)
+		if err := infraClusterClient.Create(ctx, vm); err != nil {
+			return fmt.Errorf("create VM in infrastructure Cluster: %s", err)
 		}
 		r.Recorder.Eventf(machine, corev1.EventTypeNormal, "CreatedVM", "Create VM %q", vm.Name)
+		return fmt.Errorf("VM %q not found", vm.Name)
```

Comment:
This should be removed?

Reply:
Requeue the object by returning an error.